packages/block-library/src/gallery/edit.js

```diff
@@ -66,144 +64,105 @@ const MOBILE_CONTROL_PROPS_RANGE_CONTROL = Platform.select( {
 
 function GalleryEdit( props ) {
 	const {
+		setAttributes,
 		attributes,
+		clientId,
+		noticeOperations,
 		className,
 		isSelected,
 		noticeUI,
-		noticeOperations,
-		mediaUpload,
-		imageSizes,
-		resizedImages,
-		onFocus,
+		insertBlocksAfter,
 	} = props;
+
 	const {
-		columns = defaultColumnsNumber( attributes ),
-		imageCrop,
-		images,
 		linkTo,
+		columns = defaultColumnsNumber( images ),
 		sizeSlug,
+		imageUploads,
+		imageCrop,
 	} = attributes;
-	const [ selectedImage, setSelectedImage ] = useState();
-	const [ attachmentCaptions, setAttachmentCaptions ] = useState();
 
-	function setAttributes( newAttrs ) {
-		if ( newAttrs.ids ) {
-			throw new Error(
-				'The "ids" attribute should not be changed directly. It is managed automatically when "images" attribute changes'
-			);
-		}
+	const [ images, setImages ] = useState( [] );
 
-		if ( newAttrs.images ) {
-			newAttrs = {
-				...newAttrs,
-				// Unlike images[ n ].id which is a string, always ensure the
-				// ids array contains numbers as per its attribute type.
-				ids: map( newAttrs.images, ( { id } ) => parseInt( id, 10 ) ),
-			};
-		}
-
-		props.setAttributes( newAttrs );
-	}
+	const getBlock = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getBlock;
+	} );
```

Comment:
This `useSelect` call is also missing a dependency array.
```suggestion
	const { getBlock } = useSelect( ( select ) => select( 'core/block-editor' ), [] );
```

Reply:
Fixed.